content/docs/learning-guide/tutorials/custom-tools/shape-example-cpp.md

```diff
@@ -0,0 +1,383 @@
+---
+linkTitle: Create a Custom Shape Tool in C++
+title: Create a Custom Shape Tool in C++ for Open 3D Engine
+description: Learn to create a custom shape tool in C++ to extend Open 3D Engine (O3DE).
+weight: 200
+toc: true
+---
+
+In this tutorial, you'll create a custom tool, **Shape Example**, in C++ that extends **Open 3D Engine (O3DE)**. The **Shape Example** tool allows you to create entities with a Shape component and configure their component properties. You'll learn how to use the `CppToolGem` template and practice C++ development with the Tools UI API and other O3DE APIs.
+
+This tutorial is based off of the **ShapeExample** Gem in [`o3de/sample-code-gems` repository](https://github.com/o3de/sample-code-gems/tree/main/cpp_gems/ShapeExample). You can reference the ShapeExample Gem sample as you follow along this tutorial.
+
+By the end of this tutorial, you'll be able to create your own tools that extends **O3DE Editor**.
+
+
+## Create a tool with the `CppToolGem` template
+
+The `CppToolGem` template contains a basic framework to create a dockable widget in O3DE Editor.
+
+1. Create a Gem using the **O3DE CLI (`o3de`)** script in `<engine>/scripts`. Specify the name `ShapeExample` by using the `--gem-name` option and the template `CppToolGem` by using the `--template-name` option. By default, this creates the Gem directory in `<user>/.o3de/Gems` -- or you can specify a location by using the `--gem-path` option. Depending on the location, this automatically registers the Gem to either the `.o3de` registry, engine, or project.
+
+    ```cmd
+    scripts/o3de create-gem --gem-name ShapeExample --template-name CppToolGem
+    ```
+
+2. Add the Gem in your project by using the `o3de` script.
+
+    ```cmd
+    scripts/o3de enable-gem -gn CppToolGem -pp <project-path>
+    ```
+
+    Or, enable the Gem using the Project Manager (refer to [Adding and Removing Gems in a Project](/docs/user-guide/project-config/add-remove-gems.md)).
+
+3. Build the project by using the `o3de` script (refer to [Build a project](https://o3de.org/docs/user-guide/build/configure-and-build/#build-a-project)). Or, use the Project Manager (refer to the **Build** action in the [Project Manager](https://o3de.org/docs/user-guide/project-config/project-manager/)) page.
+
+4. Open O3DE Editor for your project.
+
+5. Open the tool by selecting **Tools > Examples**.
+
```

Comment:
Could we add a screenshot of where this is on the File Menu? Just a nice and simple one. We don't need to screen shot the whole editor. Just the top left section of it with the File Menu open.